add content about lazy uniqueness check (#11166)
pingcap/docs-cn: error-codes.md

    完整的报错信息为 `ERROR 8005 (HY000) : Write Conflict, txnStartTS is stale`。

>     事务在 TiDB 中遇到了写入冲突。

Reviewer: 如果发生改错误，用户要如何处理呢

Author: ~~接受失败,~~ 调整业务/应用侧重试。

Reviewer: 意思是，当遇到了写入冲突后，用户需要去查看 TiDB 中的最新数据，改写错误对应的 SQL 语句，然后重试吗？

Author: 不一定要查看数据或者改写 SQL，看应用的逻辑吧。

    非事务 DML 语句的一个 batch 报错，语句中止，请参考[非事务 DML 语句](/non-transactional-dml.md)

* Error Number: 8147

>    当 [`tidb_constraint_check_in_place_pessimistic`](/system-variables.md#tidb_constraint_check_in_place_pessimistic) 设置为 0 时，悲观事务中的惰性唯一约束检查可能无法顺利进行，导致 SQL 语句执行报错并中止当前事务。

Reviewer: ```suggestion
   当 [`tidb_constraint_check_in_place_pessimistic`](/system-variables.md#tidb_constraint_check_in_place_pessimistic-从-v630-版本开始引入) 设置为 0 时，悲观事务中的唯一约束检查可能无法通过，导致 SQL 语句执行报错并中止当前事务。
```

* Error Number: 9007

    报错信息以 `ERROR 9007 (HY000) : Write conflict` 开头。

>     如果报错信息中含有 "reason=LazyUniquenessCheck"，说明是悲观事务并且设置了 `@@tidb_constraint_check_in_place_pessimistic=0`，业务中存在唯一索引上的写冲突，此时悲观事务不能保证执行成功。可以在应用测重试事务，或将该变量设置成 `1` 绕过。详见[约束](/constraints.md#悲观事务)。

Reviewer: ```suggestion
    如果报错信息中含有 "reason=LazyUniquenessCheck"，说明是悲观事务并且设置了 `@@tidb_constraint_check_in_place_pessimistic=OFF`，业务中存在唯一索引上的写冲突，此时悲观事务不能保证执行成功。可以在应用测重试事务，或将该变量设置成 `ON` 绕过。详见[约束](/constraints.md#悲观事务)。
```

* Error Number: 9007

    报错信息以 `ERROR 9007 (HY000) : Write conflict` 开头。

>     如果报错信息中含有 "reason=LazyUniquenessCheck"，说明是悲观事务并且设置了 `@@tidb_constraint_check_in_place_pessimistic=0`，业务中存在唯一索引上的写冲突，可以在应用测重试事务，或将该变量设置成 `1` 绕过。

Reviewer 2: 可以提一下关闭之后并不能保证事务执行成功，业务语句在唯一索引加锁语句会返回 duplicate entry 报错，或者 link 一下下面的例子或者详细说明。一种情况是开了 lazy check 之后（假如没有仔细阅读功能文档），遇到错误先从错误码找到这里。

    完整的报错信息为 `ERROR 9007 (HY000) : Write Conflict`。

>     如果报错信息中含有 "reason=LazyUniquenessCheck"，说明是悲观事务并且设置了 `@@tidb_constraint_check_in_place_pessimistic=off`，业务中存在唯一索引上的写冲突，可以在应用测重试事务，或将该变量设置成 `on` 绕过。

Author: 不会。8147 是在事务执行中报的，write conflict 一般是在 commit 时才会报给用户。遇到 8147，或者 reason=LazyUniquenessCheck 的 9007，说明是悲观事务并且设置了 `@@tidb_constraint_check_in_place_pessimistic=off`。

    完整的报错信息为 `ERROR 9007 (HY000) : Write Conflict`。

>     如果报错信息中含有 "reason=LazyUniquenessCheck"，说明是悲观事务并且设置了 `@@tidb_constraint_check_in_place_pessimistic=off`，业务中存在唯一索引上的写冲突，可以在应用测重试事务，或将该变量设置成 `on` 绕过。

Reviewer: ```suggestion
    如果报错信息中含有 "reason=LazyUniquenessCheck"，说明是悲观事务并且设置了 `@@tidb_constraint_check_in_place_pessimistic=off`，业务中存在唯一索引上的写冲突，可以在应用测重试事务，或将该变量设置成 `1` 绕过。
```
